pkg/k8s/volumes/crud.go

```diff
@@ -71,12 +71,21 @@ func CreateForDev(ctx context.Context, dev *model.Dev, c *kubernetes.Clientset,
 		pvc.Spec.VolumeName = k8Volume.Spec.VolumeName
 		_, err = vClient.Update(ctx, pvc, metav1.UpdateOptions{})
 		if err != nil {
-			return fmt.Errorf("error updating kubernetes volume claim: %s", err)
+			if !isDynamicallyProvisionedPVCError(err, pvc.Name) {
+				return fmt.Errorf("error updating kubernetes volume claim: %s", err)
+			}
+			oktetoLog.Debug("could not update pvc in namespace: %w", dev.Namespace, err)
+			oktetoLog.Warning("Error while updating the dev volume: try running 'okteto down -v' and 'okteto up', or talk to your administrator (the PVC's storage class must support 'allowVolumeExpansion' to be able to upscale dev volumes)")
 		}
 	}
 	return nil
 }
 
+func isDynamicallyProvisionedPVCError(err error, pvcName string) bool {
+	errorString := fmt.Sprintf("persistentvolumeclaims \"%s\" is forbidden: only dynamically provisioned pvc can be resized and the storageclass that provisions the pvc must support resize", pvcName)
```

Comment:
Is there a way to get this string (or the error) from the kube client sdk? It's very easy for this to change and break the error handling(we saw this with the volume snapshots recently)

Reply:
probably we can use `k8s.io/apimachinery/pkg/api/errors`. We're already getting some k8s errors from it

Reply:
The only thing I found that identifies this error is the message. The error only has these fields set:

```yaml
Status: "Failure"
Message: "persistentvolumeclaims \"web-okteto\" is forbidden: only dynamically provisioned pvc can be resized and the storageclass that provisions the pvc must support resize"
Reason: "Forbidden"
Details.Kind: "persistentvolumeclaims"
Code: 403
```

Reading from https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go, I couldn't find anything related to this specific error

Reply:
@teresaromero From what I've seen, there are other `Forbidden` errors that we don't want to catch here. Catching all `Forbidden` errors could result to wrong printing to the user. 

For example, we could catch the error `error when creating ".": persistentvolumeclaims "wp-pv-claim" is forbidden: exceeded quota` and we'd be printing to the user the message about volume expansion

Reply:
Can we use the cause error field to filter them?

Reply:
@jLopezbarb If you are talking about the `Causes` slice, it is empty.

Here's the error with all its fields expanded:
![image](https://user-images.githubusercontent.com/49416765/191025048-651fcdad-a986-4fbe-9353-90a5948b74de.png)

Reply:
Then I don't know how we can get the message without rewriting the message 😵‍💫